plugins/jenkins-backend/config.d.ts

```diff
@@ -38,11 +38,16 @@ export interface Config {
        * instance.
        */
       name: string;
-
       baseUrl: string;
       username: string;
       /** @visibility secret */
       apiKey: string;
+      extraRequestHeaders?:
+        | {
+            /** @visibility secret */
+            Authorization?: string;
+          }
+        | Record<string, string>;
```

Comment:
this should be `&` - otherwise the literal object still won't match e.g. `{ Authorization: 'Basic xx', OtherThing: 'foo' }` right?

Reply:
I check that with typescript. And if we have `Authorization ` optional with `?` it doesn't matter if we choose `|` or `&`. If it's not optional and we use `&` it breaks, with `|` it still works.

But thinking about it, using `partial` would probably show more intention.

See https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDeAoK7YFdgAsD2ATgJYBeAhsEXgHYD8AXFAM7DHUDmA3GhmdvsXKUaDZqyIduAXxQpQkKACEEUAEoQAxoQAmAHhZt2AGjGGAfNxQB6K1ABSmFlABmhKADcyAGyLaKVahQtaic+XAB5ai8QRjhEVAwsXEJSfxpGACJFMiYiDQyjHnQAI0xSzK9vMgyUKUtgp0xqAPCCGH5YqAAfJRUEjHbkwTTqTOzc-MK62QbgKCaWgkUyzE6e5XiiqFLyqAzKyprpoJpG5ppWtd7NxMGBVOFRvfG8gq2d1b2D6tr607mFjQYNRtHdOgAya7ILZ3FJCAJjHKvKZ-EIA87UYHaZa7OCQjbQxIfCpVI6os4BLEQqH9dCw4aPRETN5ElYkw6-WQ2KAABTIBEo3lk8mgAGE+jD+HCRowDBIuFswvd4elTPLuIkANoAawgMTVHAAurLxJJasLwNAACIqPkCojeXSiswzf5QMD8wVeMFQG03AZShkI55Iyack5o92eh1eHGfP2EjDEr6k8OzKP27yMBO0pLKmUh5mFVm7fap6boIA